src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,110 @@
+from urllib import urlopen
+import validators
+from datetime import datetime, timedelta
+
+class FullActivityDumpMessageTestcase(sas_testcase.SasTestCase):
+
+    def setUp(self):
+        self._sas, self._sas_admin = sas.GetTestingSas()
+        self._sas_admin.Reset()
+
+    def tearDown(self):
+        pass
+
+    @winnforum_testcase
+    def test_WINNF_FT_S_FAD_1(self):
+        """This test verifies that a SAS Under Test can successfully respond to a full activity dump request from a SAS Test Harness
+
+        Response Code should be 200
+        """
+        # register devices
+        device_a = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+        device_c = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_c.json')))
+        self._sas_admin.InjectFccId({'fccId': device_c['fccId']})
+        request = {'registrationRequest': [device_a, device_c]}
+        response = self._sas.Registration(request)['registrationResponse']
+        # Check registration response
+        cbsd_ids = []
+        for resp in response:
+            self.assertEqual(resp['response']['responseCode'], 0)
+            cbsd_ids.append(resp['cbsdId'])
+        del request, response
+
+        # Create grant requests
+        grant_0 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_1 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_0['cbsdId'] = cbsd_ids[0]
+        grant_1['cbsdId'] = cbsd_ids[1]
+        # Request for non-overlapping frequency spectrum
+        grant_0['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3620000000.0,
+            'highFrequency': 3630000000.0
+        }
+        grant_1['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3640000000.0,
+            'highFrequency': 3650000000.0
+        }
+        request = {'grantRequest': [grant_0, grant_1]}
+        # Send grant requests
+        response = self._sas.Grant(request)['grantResponse']
+        # Check registration response
+        grant_ids = []
+        for resp in response:
+            self.assertEqual(resp['response']['responseCode'], 0)
+            grant_ids.append(resp['grantId'])
+        del request, response
+
+        # Trigger the SAS to generate the activity dump
+        self._sas_admin.TriggerFullActivityDump()
+
+        # Get dump 
+        response = self._sas.GetFullActivityDump()
+        # Verify the response with  FullActivityDump.schema.json Object schema
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        generation_date_time = datetime.strptime(response['generationDateTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+        if generation_date_time >= (datetime.utcnow() - timedelta(days=7)):
```

Comment:
I'm not sure that this is a useful condition to check: if set correctly, the generation timestamp should be ~Now(); if set incorrectly, the SAS might accidentally pass this test. If you really want to check the generation timestamp for correctness (which is not written in the test spec), it's better to do it directly.